src/main/scala/com/databricks/labs/overwatch/pipeline/Restore.scala

```diff
@@ -0,0 +1,129 @@
+package com.databricks.labs.overwatch.pipeline
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.env.{Database, Workspace}
+import com.databricks.labs.overwatch.pipeline.Restore.logger
+import com.databricks.labs.overwatch.utils.Helpers.{logger, removeTrailingSlashes}
+import com.databricks.labs.overwatch.utils.{BadConfigException, Config, Helpers, SparkSessionWrapper}
+import org.apache.log4j.{Level, Logger}
+
+import java.io.FileNotFoundException
+
+
+/**
+ * Class for restoring from Migrated location to Source Location incase of some failure happens in Source Side.
+ * @param _sourceETLDB        ETL Database for Souce from where Restore need to be done.
+ * @param _targetPrefix       Target Path for where Restore would be done
+ * @param _workspace          Workspace from where Restore would be performed
+ * @param _database           Workspace Database Name.
+ * @param _config             Source Workspace Config.
+ */
+
+class Restore (_sourceETLDB: String, _targetPrefix: String, _workspace: Workspace, _database: Database, _config: Config)
+  extends Pipeline(_workspace, _database, _config) {
+
+  import spark.implicits._
+
+  private val targetPrefix = removeTrailingSlashes(_targetPrefix)
+  private val workSpace = _workspace
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  private[overwatch] def restore(
+                                  allTargets: Array[PipelineTable],
+                                  cloneLevel: String = "DEEP",
+                                ): Unit = {
+    val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+    require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+      s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+    try {
+      dbutils.fs.ls(targetPrefix)
+    } catch {
+      case e: Exception => dbutils.fs.mkdirs(targetPrefix)
```

Comment:
we can use Helpers.pathExists(targetPrefix) for the same.

Reply:
Done